### PR TITLE
[Impeller] move aiks text tests to DL.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Part of https://github.com/flutter/flutter/issues/142054

### CODE REVIEW

**Comment by jonahwilliams** [comment]
I'm looking at an issue with SiblingSaveLayerBoundsAreRespected . It looks like the saveLayer bounds are always treated as tight with aiks, but are advisory in display list - leading to the saveLayer expanding instead of clipping. Since this is the current behavior this PR isn't breaking anything there, but perhaps I should disable/remove the test as it doesn't actually demonstrate what the expected rendering is?

**Comment by flar** [comment]
> I'm looking at an issue with SiblingSaveLayerBoundsAreRespected . It looks like the saveLayer bounds are always treated as tight with aiks, but are advisory in display list - leading to the saveLayer expanding instead of clipping. Since this is the current behavior this PR isn't breaking anything there, but perhaps I should disable/remove the test as it doesn't actually demonstrate what the expected rendering is?

The intent is to include the tight bounds with the user bounds as a clip: https://github.com/flutter/engine/blob/206e86ee8a40ad28233338b7a99a2b083cdce85f/display_list/dl_builder.cc#L617

File: impeller/display_list/aiks_dl_basic_unittests.cc [hunk 1]
```diff
@@ -1067,5 +1069,445 @@ TEST_P(AiksTest, EmptySaveLayerRendersWithClear) {
...
+  paint.setColor(DlColor::kRed());
+  builder.DrawRect(rect, paint);
+
+  ASSERT_TRUE(OpenPlaygroundHere(builder.Build()));
+}
+
+TEST_P(AiksTest, SiblingSaveLayerBoundsAreRespected) {
+  DisplayListBuilder builder;
+  DlPaint paint;
+  SkRect rect = SkRect::MakeXYWH(0, 0, 1000, 1000);
```
Comment (flar) [inline impeller/display_list/aiks_dl_basic_unittests.cc:1374]: I don't remember where it caused a problem but in some tool I've used to look at goldens a transparent background with black stuff drawn on it just looks like a giant field of black, so I usually paint a background color in my tests...

**Comment by jonahwilliams** [comment]
Ahh found the bug!

**Comment by jonahwilliams** [comment]
The rrect changes are a bit odd too, but I think the Skia RRect constructor must be different from the impeller version. Otherwise ready for review, I've left the untriaged goldens.

**Comment by jonahwilliams** [comment]
Friendly ping @flar  :)

**Review by chinmaygarde — APPROVED** [review]

**Comment by flar** [comment]
Sorry, I was still looking at why the goldens changed. I don't trust that.

**Comment by flar** [comment]
Is there any way to load up the changed goldens again?

**Comment by chinmaygarde** [comment]
Hmm, not that I know of. Perhaps you can create a revert PR (but not submit it) to show the reverse change to introspect.

**Pull request merged**